### DISCUSSION TITLE
Módulo Flutter em App Nativo

### DISCUSSION THREAD

**Discussion by Zeca-dev** [Flutterando/forum#111] [Perguntas e Respostas] [answered]
Estou fazendo uma poc para criação de módulo flutter dentro de um aplicativo nativo existente. Já consegui navegar para o módulo de algumas formas, mas todas tem o mesmo problema:

Quando eu navego para uma rota específica ("/page2") e volto. Em vez de eu sair do módulo flutter e voltar para o nativo, eu acabo voltando para a rota inicial do módulo ("/"), sendo que eu não quero exibir essa página. 

```
 routes: {
        "/": (context) => const MyHomePage(title: 'Home'),
        "/page2": (context) => const Page2(),
      },
```

Já tentei usar o dispose da Page2() para dar mais um pop, porém sem sucesso.
Como faço para fechar o módulo todo quando sair de uma rota?

Obrigado.

**Comment by jonathanBAugusto** [comment, marked as answer]
Se voce usou um Navigator.push para ir para a Rota 2, então é bem provavel que `/` esteja na stack ainda. O que tu pode fazer é executar `Navigator.pop(context);` antes de chavar o push, ou usar a `Navigator.popAndPushNamed(...);` que ja vai resolver. Isso claro se voce navegou da home para a page2

**Reply by Zeca-dev** [reply]
Acabei testando com o Modular e funcionou. As demais formas não deram certo.
